libraries/rib-base/src/main/java/com/badoo/ribs/android/integrationpoint/FragmentIntegrationPoint.kt

```diff
@@ -0,0 +1,70 @@
+package com.badoo.ribs.android.integrationpoint
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import com.badoo.ribs.android.AndroidRibViewHost
+import com.badoo.ribs.android.activitystarter.ActivityBoundary
+import com.badoo.ribs.android.activitystarter.ActivityStarter
+import com.badoo.ribs.android.dialog.DefaultDialogLauncher
+import com.badoo.ribs.android.dialog.DialogLauncher
+import com.badoo.ribs.android.permissionrequester.PermissionRequestBoundary
+import com.badoo.ribs.android.permissionrequester.PermissionRequester
+import com.badoo.ribs.android.store.RetainedInstanceStoreViewModel
+
+open class FragmentIntegrationPoint(
+    private val fragment: Fragment,
+    savedInstanceState: Bundle?,
+    private val ribHostViewProvider: (View?) -> ViewGroup? = {
+        if (it != null) {
+            require(it is ViewGroup)
+            it
+        } else {
+            null
+        }
+    }
+) : IntegrationPoint(
+    lifecycleOwner = fragment,
+    viewLifecycleOwner = fragment.viewLifecycleOwnerLiveData,
+    savedInstanceState = savedInstanceState,
+    rootViewHostFactory = { ribHostViewProvider(fragment.requireView())?.let(::AndroidRibViewHost) }
+) {
+    private val activityBoundary = ActivityBoundary(fragment, requestCodeRegistry)
+    private val permissionRequestBoundary = PermissionRequestBoundary(fragment, requestCodeRegistry)
+    private val retainedInstanceViewModel by fragment.viewModels<RetainedInstanceStoreViewModel>()
+
+    override val activityStarter: ActivityStarter
+        get() = activityBoundary
+
+    override val permissionRequester: PermissionRequester
+        get() = permissionRequestBoundary
+
+    override val dialogLauncher: DialogLauncher =
+        DefaultDialogLauncher(fragment.requireContext(), fragment.lifecycle)
+
+    override val isFinishing: Boolean
+        get() = retainedInstanceViewModel.isCleared
+
+    init {
+        retainedInstanceViewModel // initialize ViewModel
+    }
+
+    override fun handleUpNavigation() {
+        fragment.requireActivity().onNavigateUp()
```

Comment:
why navigation Up, not onBackPressed?

Reply:
Here we actually mean up navigation. Back press handling done separately.